Karl/test.cc

```diff
@@ -744,6 +744,88 @@ TEST(Karl, ChangeKarlWhichClaireReportsTo) {
   }
 }
 
+TEST(Karl, ChangeDependenciesInClaire) {
+  current::time::ResetToZero();
+
+  const auto stream_file_remover = current::FileSystem::ScopedRmFile(FLAGS_karl_test_stream_persistence_file);
+  const auto storage_file_remover = current::FileSystem::ScopedRmFile(FLAGS_karl_test_storage_persistence_file);
+  const unittest_karl_t karl(UnittestKarlParameters());
+  const current::karl::Locator karl_locator(Printf("http://localhost:%d/", FLAGS_karl_test_keepalives_port));
+  const uint16_t claire_port = PickPortForUnitTest();
+  current::karl::Claire claire(karl_locator, "unittest", claire_port, {"http://127.0.0.1:12345"});
+  claire.Register(nullptr, true);
+
+  // Karl sees single `claire`'s dependency on `127.0.0.1:12345` as unresolved.
+  {
+    unittest_karl_status_t status;
+    const auto body = HTTP(GET(Printf("http://localhost:%d?full&from=0", FLAGS_karl_test_fleet_view_port))).body;
+    ASSERT_NO_THROW(status = ParseJSON<unittest_karl_status_t>(body)) << body;
+#if 0
+    ASSERT_NO_THROW(
+        status = ParseJSON<unittest_karl_status_t>(
+            HTTP(GET(Printf("http://localhost:%d?from=0", FLAGS_karl_test_fleet_view_port))).body));
+#endif
+    EXPECT_EQ(1u, status.machines.size()) << JSON(status);
+    ASSERT_TRUE(status.machines.count("127.0.0.1")) << JSON(status);
+    auto& server = status.machines["127.0.0.1"];
+    auto& per_ip_services = server.services;
+    ASSERT_EQ(1u, per_ip_services.size());
+    auto& claire_service_info = per_ip_services[claire.Codename()];
+    ASSERT_EQ("unittest", claire_service_info.service);
+    EXPECT_EQ(0u, claire_service_info.dependencies.size());
```

Comment:
`EXPECT_EQ` will print the actual count of elements, which I consider more informative. What do you think?